client/src/components/ScheduledAppointments/ScheduledAppointmentsTab.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from "react";
+import { Grid } from "@material-ui/core";
+import axios from "axios";
+import { parseISO, startOfToday, compareAsc } from "date-fns";
+
+import { useUserData } from "../../providers/Context";
+import IndividualAppointment from "./IndividualAppointment";
+import YourAppointments from "./YourAppointments";
+
+const ScheduledAppointmentsTab = () => {
+  const userData = useUserData();
+  const { _id: userId } = userData;
+
+  const [upcomingAppointments, setupcomingAppointments] = useState([]);
+  const [pastAppointments, setPastAppointments] = useState([]);
+
+  useEffect(() => {
+    axios.get("/api/all-appointments").then(({ data }) => {
+      // data: [{_id, email, eventId, hostId, name, time, timezone, duration},...]
+
+      const upcomingAppointments = [];
+      const pastAppointments = [];
+
+      data.forEach((appointment) => {
+        if (appointmentIsUpcoming(appointment.time)) {
+          upcomingAppointments.push(appointment);
+        } else {
+          pastAppointments.push(appointment);
+        }
+      });
+
+      setupcomingAppointments(upcomingAppointments);
+      setPastAppointments(pastAppointments);
+    });
+  }, [userId]);
+  return (
+    <Grid container direction="column">
+      <YourAppointments
+        upcoming={true}
+        empty={upcomingAppointments.length === 0}
+      />
+
+      {upcomingAppointments.length !== 0 &&
+        upcomingAppointments.map((appointment) => (
+          <IndividualAppointment
+            key={appointment._id}
+            appointmentId={appointment._id} //this is for child access
+            upcoming={true}
+            attendeeName={appointment.name}
+            time={appointment.time}
+            duration={appointment.duration}
+            attendeeEmail={appointment.email}
+            attendeeTimezone={appointment.timezone}
+          />
+        ))}
+
+      <YourAppointments
+        upcoming={false}
+        empty={pastAppointments.length === 0}
+      />
+
+      {pastAppointments.length !== 0 &&
+        pastAppointments.map((appointment) => (
+          <IndividualAppointment
+            key={appointment._id}
+            appointmentId={appointment._id} //this is for child access
+            upcoming={false}
+            attendeeName={appointment.name}
+            time={appointment.time}
+            duration={appointment.duration}
+            attendeeEmail={appointment.email}
+            attendeeTimezone={appointment.timezone}
+          />
+        ))}
+    </Grid>
+  );
+};
+
+const appointmentIsUpcoming = (appointmentDateInISOString) => {
+  // upcoming === true indicate that the appointment is either today or tomorrow onward
+  const today = startOfToday();
```

Comment:
Should we compare to the start of today or to the current time?

Reply:
I might have compared it to the start of today to keep things simple. But yeah it makes more sense to compare to the current time.

Reply:
It depends... on what we consider upcoming(tomorrow vs in a couple hours) and also how `compareAsc` calculates the comparison.

Reply:
Yeah Matt we definitely should discuss that. I have set up the function so I can easily adjust what we consider as upcoming. The component only need to know whether it's true or false lol.